cypress/e2e/gui-tests/home.cy.js

```diff
@@ -0,0 +1,253 @@
+describe('Home Page Tests', () => {
+    beforeEach(() => {
+        cy.visit('/')
+    })
+
+    it('Renders the header with an h1 and theme toggle', () => {
+        cy.get('h1').should('exist').and('have.text', 'EngageSphere');
```

Comment:
Veja uma opção mais curta e simples:

```suggestion
        cy.contains('h1', 'EngageSphere').should('be.visible');
```